sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -354,27 +367,29 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
         }
 
         def createCartesianProduct() = {
-          if (joinType.isInstanceOf[InnerLike]) {
+          if (joinType.isInstanceOf[InnerLike] && !hintToNotBroadcastAndReplicate(hint)) {
```

Comment:
Is `CartesianProduct` related to broadcast?

Reply:
It is related to replication.

Reply:
The new hint prohibits both broadcasting and replication so it applies to 3 cases:
- Broadcast hash
- Cartesian product
- BNLJ build side

Reply:
It would be nice to update this to the pr description too.